teamprojekt_competition_server/server/protocol.py

```diff
@@ -53,15 +55,27 @@ def connectionLost(self, reason):
 
         return super().connectionLost(reason)
 
-    def get_token(self, return_callback: Callable[[str], None]) -> None:
+    def get_token(
+        self,
+        return_callback: Callable[[str], None],
+        delete_callback: Callable[[], None],
+    ) -> None:
         """get token from client to authenticate
 
         Args:
             game (Game): game that starts
         """
-        self.callRemote(Auth).addCallback(
-            callback=lambda res: return_callback(res["token"])
-        )
+
+        def callback(res):
+            if res["version"] == VERSION:
+                return_callback(res["token"])
+            else:
+                log.error("Client with wrong version tried to authenticate.")
+                delete_callback()
+                # self.transport.loseConnection()

```

Comment:
If the Version is incompatible we should disconnect, the introduced delete is unneccesarry due to it handeles the same functionality as loseConnection()